src/App.tsx

```
@@ -1,61 +1,157 @@
+import { Todo } from './types/Todo';
 
 export const App = () => {
+  const getUserById = (userId: number): User => {
+    const userById = usersFromServer.find((user: User) => {
+      return user.id === userId;
+    });
+
+    if (!userById) {
+      return null;
+    }
+
+    return userById;
+  };
+
+  const todosInitial: Todo[] = [];
+
+  todosFromServer.map((todo: Todo) => {
+    todosInitial.push({
+      ...todo,
+      user: getUserById(todo.id),
+    });
+  });
+
+  const [todos, setTodos] = useState<Todo[]>([...todosInitial]);
+
+  const [title, setTitle] = useState<string>('');
+  const [titleError, setTitleError] = useState<boolean>(false);
+
+  const [option, setOption] = useState<string>('0');
+  const [optionError, setOptionError] = useState<boolean>(false);
+
+  const onReset = () => {
+    setTitle('');
+    setOption('0');
+    setTitleError(false);
+    setOptionError(false);
+  };
+
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  const handleFormReset = (event: React.FormEvent) => {
+    onReset();
+  };
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos([...todos, newTodo]);
+  };
+
+  const handleFormSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title || title === '') {
+      setTitleError(true);
+    }
+
+    if (!option || option === '0') {
+      setOptionError(true);
+
+      return;
+    }
+
+    if (title !== '' && option !== '0') {
+      addTodo({
+        id: todos.length,
+        title: title,
+        completed: false,
+        userId: Number(option),
+        user: getUserById(Number(option)),
+      });
+    }
+
+    onReset();
+  };
+
+  const handleChangeInputTitle = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    const { value } = event.target;
+
+    setTitle(value);
+  };
+
+  const handleChangeSelect = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setOption(value);
+  };
```

To follow the requirement of hiding the error message on input change, you should add `setOptionError(false)` inside this handler.